infrastructure/lib/constructs/snsMonitor.ts

```diff
@@ -0,0 +1,116 @@
+import { Construct } from 'constructs';
+import { Alarm } from 'aws-cdk-lib/aws-cloudwatch';
+import * as cloudwatch from "aws-cdk-lib/aws-cloudwatch";
+import * as sns from "aws-cdk-lib/aws-sns";
+import * as subscriptions from "aws-cdk-lib/aws-sns-subscriptions";
+import * as actions from "aws-cdk-lib/aws-cloudwatch-actions";
+import { Canary } from 'aws-cdk-lib/aws-synthetics';
+import {OpenSearchLambda} from "./lambda";
+
+interface SnsMonitorsProps {
+    readonly region: string;
+    readonly accountId: string;
+    readonly stepFunctionSnsAlarms?: Array<{ alertName: string, stateMachineName: string }>;
+    readonly canaryAlarms?: Array<{ alertName: string, canary: Canary }>;
+    readonly alarmNameSpace: string;
+    readonly snsTopic: string;
+    readonly slackLambda: OpenSearchLambda;
+}
+
+export class SnsMonitors extends Construct {
+    private readonly region: string;
```

Comment:
NIT: Try to see if you can seperate the `stepFunctionSnsAlarms` and `canaryAlarms` into new files and extend the common methods.